pymodbus/transaction.py

```diff
@@ -463,13 +468,17 @@ def _recv(self, expected_response_length, full) -> bytes:  # noqa: C901
                         expected_response_length -= min_size
                         total = expected_response_length + min_size
                 else:
+                    if exception_length is None:
+                        exception_length = 0
                     expected_response_length = exception_length - min_size
                     total = expected_response_length + min_size
             else:
                 total = expected_response_length
         else:
             read_min = b""
             total = expected_response_length
+        if expected_response_length is None:
```

Comment:
Not covered by test.

Reply:
This was flagged by the type-checker, `expected_response_length` can't be `None` as we pass `expected_response_length` to `self.client.framer.recvPacket` which expects an integer.

Reply:
Did you read my comment ?

Reply:
None can be converted to a 0 without problems.

Reply:
> None can be converted to a 0 without problems.

Updated to no longer convert to 0.